staging/src/k8s.io/component-base/metrics/testutil/metrics_test.go

```diff
@@ -512,3 +516,35 @@ func TestHistogramVec_Validate(t *testing.T) {
 		})
 	}
 }
+
+func TestGetHistogramVecFromGatherer(t *testing.T) {
+	// HistogramVec has two label defined.
+	labels := []string{"label1", "label2"}
+	// expect to fetch all the metrics with the pair of {label1=value1}
+	lvMap := map[string]string{"label1": "value1"}
+	HistogramOpts := &metrics.HistogramOpts{
+		Namespace: "namespace",
+		Name:      "metric_test_name",
+		Subsystem: "subsystem",
+		Help:      "histogram help message",
+		Buckets:   prometheus.DefBuckets,
+	}
+	vec := metrics.NewHistogramVec(HistogramOpts, labels)
+	legacyregistry.MustRegister(vec)
+	// Observe two metrics with same value for label1 but different value of label2.
+	vec.WithLabelValues("value1", "value2-0").Observe(1.5)
+	vec.WithLabelValues("value1", "value2-1").Observe(3.5)
```

Comment:
add a sample with a different value for label1

Reply:
Done